packages/core/src/components/HierarchyPicker/Hierarchy/HierarchyList.tsx

```diff
@@ -0,0 +1,287 @@
+          styles.pane,
+          isNested && styles.pane_nested,
+          isNested && !verticallyAlign && styles.pane_verticallyOffset,
+          styles.aside,
+          { width },
+        )}
+      >
+        <button
+          {...css(styles.asideButton)}
+          onClick={() => onItemPicked([...parents, item.name])}
+          tabIndex={-1}
+          type="button"
+        >
+          <ItemDescription item={item} />
+        </button>
+      </aside>
+    );
+  }
+
+  render() {
+    const { focus, items, styles, parents, verticallyAlign, ...passThruProps } = this
+      .props as Required<Props & WithStylesProps>;
+
+    if (items.length === 0) {
+      return null;
+    }
+
+    const [focusName, ...focusRest] = focus;
+    const { maxHeight, width } = passThruProps;
+    const isNested = parents.length > 0;
+
+    // Track focused item to render as a sibling if vertically aligned
+    let focusedItem: ItemShape | undefined;
+    let currentSection: string | undefined;
+
+    return (
+      <>
+        <div
+          key="list"
+          {...css(
+            styles.pane,
+            verticallyAlign && styles.pane_verticallyAlign,
+            isNested && styles.pane_nested,
+            isNested && !verticallyAlign && styles.pane_verticallyOffset,
+            {
+              width,
+              maxHeight: verticallyAlign ? maxHeight : undefined,
+              zIndex: 1,
+            },
+          )}
+          ref={this.ref}
+        >
+          <ul {...css(styles.list)}>
+            {items.map((item, index) => {
+              const { name, section } = item;
+              const definition = parents.concat(name);
+              const isFocused = name === focusName;
+              const shouldRenderSection = typeof section === 'string' && section !== currentSection;
+
+              focusedItem = verticallyAlign && isFocused ? item : focusedItem;
+              currentSection = shouldRenderSection ? section : currentSection;
+
+              return (
+                <React.Fragment key={item.name}>
+                  {shouldRenderSection && index > 0 ? <li {...css(styles.divider)} /> : null}
+
+                  {shouldRenderSection && section ? (
+                    <li {...css(styles.sectionHeader)}>
+                      <Text small bold uppercased>
+                        {section}
+                      </Text>
+                    </li>
+                  ) : null}
+
+                  <li {...css(styles.row)}>
+                    <HierarchyItem
+                      {...passThruProps}
+                      onDomFocusDeeper={this.handleDomFocusDeeper}
+                      onDomFocusShallower={this.handleDomFocusShallower}
+                      item={item}
+                      definition={definition}
+                      selected={this.isChosen(definition)}
+                      focused={isFocused}
+                    />
+
+                    {!verticallyAlign && isFocused && item.items ? (
```

Comment:
`item.items.length > 0`

Reply:
the `1.0` recursive logic handles an empty array but I agree this is better 👍